py3dtilers/IfcTiler/ifcObjectGeom.py

```diff
@@ -61,258 +40,58 @@ def computeCenter(self, pointList):
-
-        listDirection = self.getDirections(self.ifcObject.ObjectPlacement)
-
-        offset = np.array(transform_matrix[3])
-        rotation = np.array(transform_matrix[:-1])
-
-        vertexList = list()
-        indexList = list()
-        for representation in representations:
-            if(representation.RepresentationType == "MappedRepresentation"):
-                representation = representation.Items[0].MappingSource.MappedRepresentation
-
-            nb_geom = 0
-            for itemGeom in representation.Items:
-                nb_geom += 1
-                if(nb_geom > 300):
-                    continue
-                indexListTemp = None
-                vertexListTemp = None
-                if(representation.RepresentationType == "Tessellation"):
-                    if(hasattr(itemGeom, 'Faces')):
-                        indexListTemp = self.parse_indexed_faces(itemGeom.Faces)
-                    elif (hasattr(itemGeom, 'CoordIndex')):
-                        indexListTemp = itemGeom.CoordIndex
-                    else:
-                        sys.exit("Géométrie de ce type non encore gérée")
-                    vertexListTemp = itemGeom.Coordinates.CoordList
-
-                elif(representation.RepresentationType == "SweptSolid" and not(itemGeom.is_a("IfcBooleanClippingResult"))):
-                    vertexListTemp, indexListTemp = self.extrudGeom(itemGeom)
-                if(vertexListTemp and indexListTemp):
-                    for index in indexListTemp:
-                        indexList.append([index[0] + len(vertexList), index[1] + len(vertexList), index[2] + len(vertexList)])
-                    for vertex in vertexListTemp:
-                        vertexList.append(np.array([vertex[0], vertex[1], vertex[2]], dtype=np.float32))
-
-        if (len(indexList) == 0):
+        try :
+            settings = geom.settings()
+            settings.set(settings.USE_WORLD_COORDS, True) #Translates and rotates the points to their world coordinates
+            settings.set(settings.SEW_SHELLS,True)
+            shape = geom.create_shape(settings,ifcObject)
+        except RuntimeError:
+            logging.error("Error while creating geom with IfcOpenShell")
             return False
 
-        for j in range(len(vertexList)):
-            vertex = vertexList[j]
-            for i in range(len(listDirection)):
-                vertex = np.dot(np.array(vertex), listDirection[i])
-                vertex = (vertex + listPosition[i])
+        vertexList = np.reshape(np.array(shape.geometry.verts),(-1,3))
+        indexList = np.reshape(np.array(shape.geometry.faces),(-1,3))
+        if(shape.geometry.materials):
+            ifc_material = shape.geometry.materials[0]
+            self.material = GlTFMaterial(rgb=[ifc_material.diffuse[0],ifc_material.diffuse[1],ifc_material.diffuse[2],ifc_material.transparency],
+                        alpha= ifc_material.transparency if ifc_material.transparency else 0,
```

Comment:
The 4th value in `rgb` and the `alpha` value are the same (and will be used to set the same field in the material). The `alpha` parameter is used only if the color is RGB and not RGBA. Maybe you should keep only one of them. 

```python
GlTFMaterial(rgb=[ifc_material.diffuse[0],ifc_material.diffuse[1],ifc_material.diffuse[2],
                        alpha= ifc_material.transparency if ifc_material.transparency else 0,
```